tls/extensions/s2n_client_supported_groups.c

```diff
@@ -78,6 +78,23 @@ static int s2n_client_supported_groups_send(struct s2n_connection *conn, struct
     return S2N_SUCCESS;
 }
 
+S2N_RESULT s2n_client_supported_groups_parse_groups_count(struct s2n_stuffer *extension, uint16_t *count)
+{
+    RESULT_ENSURE_REF(count);
+    *count = 0;
+    RESULT_ENSURE_REF(extension);
+
+    uint16_t supported_groups_list_size = 0;
+    RESULT_GUARD_POSIX(s2n_stuffer_read_uint16(extension, &supported_groups_list_size));
+
+    RESULT_ENSURE_LTE(supported_groups_list_size, s2n_stuffer_data_available(extension));
+    RESULT_ENSURE_EQ(supported_groups_list_size % sizeof(uint16_t), 0);
+
+    *count = supported_groups_list_size / 2;
```

Comment:
Nit: I don't love mixing sizeof(uint16_t) and 2 to mean the same thing so close together. We should probably choose one, or maybe use a named macro.